src/System.Net.Security/src/System/Net/Security/NegoState.cs

```diff
@@ -53,11 +53,6 @@ internal class NegoState
 
         internal NegoState(Stream innerStream, bool leaveStreamOpen)
         {
-            if (innerStream == null)
```

Comment:
The only place to create a `NegoState` is [here](https://github.com/dotnet/corefx/blob/a44c93063d24e29971891bfc2cf8b0aa8771eae9/src/System.Net.Security/src/System/Net/Security/NegotiateStream.cs#L44). Before passing the `innerStream` to `NegoState`'s ctor, the `NegotiateStream`'s base class `AuthenticatedStream` will check the stream for null value. So the check here is not needed.

Reply:
In such cases we often then assert:
```C#
Debug.Assert(innerStream != null);
```

Reply:
Good to know that! Will fix.